File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> mpp;
        vector<vector<string>> result;
        for (string str : strs) {
            string sortedStr = str;
            sort(sortedStr.begin(), sortedStr.end());
            mpp[sortedStr].push_back(str);
        }
        for (auto mp : mpp) {
            result.push_back(mp.second);
        }
        return result;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function (strs) {
    const freqMap = new Map()
    for (const str of strs) {
        const sortedStr = str.split("").sort().join("")
        if (!freqMap.has(sortedStr)) {
            freqMap.set(sortedStr, []);
        }
        freqMap.get(sortedStr).push(str);
    }

    return [...freqMap.values()];
};